cmd/junitreport/pkg/parser/interfaces.go

```diff
@@ -0,0 +1,61 @@
+package parser
+
+import (
+	"bufio"
+
+	"github.com/openshift/origin/cmd/junitreport/pkg/api"
+)
+
+// TestOutputParser knows how to parse test output to create a collection of test suites
+type TestOutputParser interface {
+	Parse(input *bufio.Scanner) (*api.TestSuites, error)
+}
+
+// TestDataParser knows how to take raw test data and either characterize it as data that
+// contains useful information or extract the useful information from it
+type TestDataParser interface {
+	// ContainsIdentifier determines if the line contains the identifier for a test
+	ContainsIdentifier(line string) bool
```

Comment:
definitely be more specific with go's duck typing.... don't want to accidentally satisfy the wrong interface